test/test_nn_quantized.py

```diff
@@ -34,6 +36,67 @@ def test_relu_api(self):
 
 
 class ModuleAPITest(TestCase):
+    @given(
+        batch_size=st.integers(1, 5),
+        in_features=st.integers(16, 32),
+        out_features=st.integers(4, 8),
+        use_bias=st.booleans(),
+    )
+    def test_dynamic_linear_api(self, batch_size, in_features, out_features, use_bias):
+        """test API functionality for nn.quantized.dynamic.Linear"""
+        W = torch.rand(out_features, in_features).float()
+        W_scale, W_zp = _calculate_dynamic_qparams(W, torch.qint8)
+        W_q = torch.quantize_linear(W, W_scale, W_zp, torch.qint8)
+        W_pack_col_offset = torch.ops.quantized.fbgemm_linear_prepack(W_q)
+        X = torch.rand(batch_size, in_features).float()
+        B = torch.rand(out_features).float() if use_bias else None
+        qlinear = nnqd.Linear(in_features, out_features)
+        qlinear._packed_weight = W_pack_col_offset
+        qlinear.bias = B
+        Z_dq = qlinear(X)
+        # Check if the module implementation matches calling the
+        # ops directly
+        Z_ref = torch.ops.quantized.fbgemm_linear_dynamic(X, W_pack_col_offset, B)
+        self.assertEqual(Z_ref, Z_dq)
+
+        # Test serialization of dynamic quantized Linear Module using state_dict
+        model_dict = qlinear.state_dict()
+        self.assertEqual(model_dict['weight'], W_q)
+        if use_bias:
+            self.assertEqual(model_dict['bias'], B)
+        with tempfile.NamedTemporaryFile() as f:
```

Comment:
if you are not going to use the file outside the `with` clause, it is better to use the `tempfile.TemporaryFile` instead -- that way, the file will be destroyed right after the `with` statement.

Reply:
Replace `tempfile.NamedTemporaryFile` with `tempfile.TemporaryFile` in this file.